src/main/java/com/powsybl/openloadflow/sensi/apiv2/SensitivityFactorConfiguration.java

```diff
@@ -0,0 +1,108 @@
+
+import java.io.IOException;
+import java.util.*;
+
+/**
+ * A set of simple factors or factor matrices associated to a contingency context.
+ *
+ * @author Geoffroy Jamgotchian <geoffroy.jamgotchian at rte-france.com>
+ */
+public final class SensitivityFactorConfiguration {
+
+    private static final String CURRENT_VERSION = "1";
+
+    private final ContingencyContext contingencyContext;
+
+    private final String contingencyId;
+
+    private final List<SimpleSensitivityFactor> simpleFactors;
+
+    private final List<MatrixSensitivityFactor> matrixFactors;
+
+    private final List<MultiVariablesSensitivityFactor> multiVarsFactors;
+
+    private SensitivityFactorConfiguration(ContingencyContext contingencyContext, String contingencyId, List<SimpleSensitivityFactor> simpleFactors,
+                                           List<MatrixSensitivityFactor> matrixFactors, List<MultiVariablesSensitivityFactor> multiVarsFactors) {
+        this.contingencyContext = contingencyContext;
+        this.contingencyId = contingencyId;
+        this.simpleFactors = new ArrayList<>(Objects.requireNonNull(simpleFactors));
+        this.matrixFactors = new ArrayList<>(Objects.requireNonNull(matrixFactors));
+        this.multiVarsFactors = new ArrayList<>(Objects.requireNonNull(multiVarsFactors));
+    }
+
+    public static SensitivityFactorConfiguration create(List<SimpleSensitivityFactor> simpleFactors,
+                                                        List<MatrixSensitivityFactor> matrixFactors,
+                                                        List<MultiVariablesSensitivityFactor> multiVarsFactors) {
+        return new SensitivityFactorConfiguration(ContingencyContext.NONE, null, simpleFactors, matrixFactors, multiVarsFactors);
+    }
+
+    public static SensitivityFactorConfiguration create() {
+        return create(Collections.emptyList(), Collections.emptyList(), Collections.emptyList());
+    }
+
+    public static SensitivityFactorConfiguration createForOneContingency(String contingencyId,
+                                                                         List<SimpleSensitivityFactor> simpleFactors,
+                                                                         List<MatrixSensitivityFactor> matrixFactors,
+                                                                         List<MultiVariablesSensitivityFactor> multiVarsFactors) {
+        Objects.requireNonNull(contingencyId);
+        return new SensitivityFactorConfiguration(ContingencyContext.ONE_CONTINGENCY, contingencyId, simpleFactors, matrixFactors, multiVarsFactors);
+    }
+
+    public static SensitivityFactorConfiguration createForOneContingency(String contingencyId) {
+        return createForOneContingency(contingencyId, Collections.emptyList(), Collections.emptyList(), Collections.emptyList());
+    }
+
+    public static SensitivityFactorConfiguration createForAllContingency(List<SimpleSensitivityFactor> simpleFactors,
```

Comment:
I am not sure I understand how this fits with the factors being mutable. From what I have understood, the results of the sensitivity analysis will be written inside the factors. But how do you write multiple values (one for each contingency) ?